contracts/src/v0.8/dev/VRFCoordinatorV2.sol

```diff
@@ -0,0 +1,435 @@
+        });
+        s_fallbackGasPrice = fallbackGasPrice;
+        s_fallbackLinkPrice = fallbackLinkPrice;
+        emit ConfigSet(minimumRequestBlockConfirmations,
+            maxConsumersPerSubscription,
+            stalenessSeconds,
+            gasAfterPaymentCalculation,
+            fallbackGasPrice,
+            fallbackLinkPrice
+        );
+    }
+
+    /**
+     * @notice read the current configuration of the coordinator.
+     */
+    function getConfig()
+    external
+    view
+    returns (
+        uint16 minimumRequestBlockConfirmations,
+        uint16 maxConsumersPerSubscription,
+        uint32 stalenessSeconds,
+        uint32 gasAfterPaymentCalculation,
+        int256 fallbackGasPrice,
+        int256 fallbackLinkPrice
+    )
+    {
+        Config memory config = s_config;
+        return (
+            config.minimumRequestBlockConfirmations,
+            config.maxConsumersPerSubscription,
+            config.stalenessSeconds,
+            config.gasAfterPaymentCalculation,
+            s_fallbackGasPrice,
+            s_fallbackLinkPrice
+        );
+    }
+
+    function requestRandomWords(
+        bytes32 keyHash,  // Corresponds to a particular offchain job which uses that key for the proofs
+        uint64  subId,
+        uint64  minimumRequestConfirmations,
+        uint64  callbackGasLimit,
+        uint64  numWords  // Desired number of random words
+    )
+    external
+    returns (uint256 requestId)
+    {
+       // One would consider doing a sanity check that the subscription has enough link here,
+       // but:
+       //   1) you'd need to either store a cached price or do a lookup which are both costly
+       //   2) set a fixed min which would be hard to estimate
+       //   3) price could fluctuate anyways between request and response, so it could still fail to fulfill
+       //   4) if it did fail to fulfill, you could simply top up the subscription and re-try and it would succeed
+       require(s_subscriptions[subId].owner != address(0), "invalid subId");
+       require(minimumRequestConfirmations >= s_config.minimumRequestBlockConfirmations, "minconfs too low");
+       bool validConsumer;
+       for (uint16 i = 0; i < s_subscriptions[subId].consumers.length; i++) {
+           if (s_subscriptions[subId].consumers[i] == msg.sender) {
+               validConsumer = true;
+               break;
+           }
+       }
+       require(validConsumer, "invalid consumer");
+       require(s_serviceAgreements[keyHash] != address(0), "must be a registered key");
+       uint256 nonce = s_nonces[keyHash][msg.sender] + 1;
+       uint256 preSeedAndRequestId = uint256(keccak256(abi.encode(keyHash, msg.sender, nonce)));
+       // Min req confirmations not needed as part of fulfillment, leave out of the commitment
+       s_callbacks[preSeedAndRequestId] = keccak256(abi.encodePacked(preSeedAndRequestId, block.number, subId, callbackGasLimit, numWords, msg.sender));
```

Comment:
Any particular reason to use `encodePacked` here instead of regular `encode`?

Reply:
its kind of a micro optimization, encodePacked wont pad which will make the cost of the hash slightly less. I took it from the v1 coordinator